modules/assim.sequential/R/sda_plotting.R

```diff
@@ -0,0 +1,546 @@
+#internal functions for plotting SDA outputs. Interactive, bias plots as well as post analysis time series both in base plotting and ggplot
+##' @param settings  pecan standard settings list.  
+##' @param t current time - int number giving the position of the current time in obs.time. 
+##' @param obs.time vector of dates of measurements
+##' @param obs.mean list of vectors of the means of observed data named by the measured date.
+##' @param obs.cov list of cov matrices of the observed data named by the measured date.
+##' @param obs list containing the mean and cov object
+##' @param X  dataframe of state variables for each ensemble
+##' @param FORECAST dataframe of state variables for each ensemble
+##' @param ANALYSIS  vector of mean of state variable after analysis
+##' @param plot.title character giving the title for post visualization ggplots
+##' @export
+interactive.plotting.sda<-function(settings,t,obs.times,obs.mean,obs.cov,obs,X,FORECAST,ANALYSIS){
+  #Defining some colors
+  t1         <- 1
+  pink       <- col2rgb("deeppink")
+  alphapink  <- rgb(pink[1], pink[2], pink[3], 180, max = 255)
+  green      <- col2rgb("green")
+  alphagreen <- rgb(green[1], green[2], green[3], 75, max = 255)
+  blue       <- col2rgb("blue")
+  alphablue  <- rgb(blue[1], blue[2], blue[3], 75, max = 255)
+  purple       <- col2rgb("purple")
+  alphapurple <- rgb(purple[1], purple[2], purple[3], 75, max = 255)
+  brown       <- col2rgb("brown")
+  alphabrown <- rgb(brown[1], brown[2], brown[3], 75, max = 255)
```

Comment:
this code chunk is repeated so many times here and in other plotting fcns, probably worth having its own fcn

Reply:
Is there a way to put it in pecan.data.viz?